android-sdk/src/main/java/ai/evolv/android_sdk/EvolvContextImpl.java

```diff
@@ -36,17 +52,75 @@ private void initialize(String uid,
         }
 
         this.uid = uid;
-        this.sid = sid;
-        Class genericRemoteContextClass = remoteContext.getClass();
-        Class genericLocalContextClass = remoteContext.getClass();
-        this.remoteContext = remoteContext != null ? helper
-                .deepCopy(remoteContext,genericRemoteContextClass) : new HashMap<>();
-        this.localContext = localContext != null ? helper
-                .deepCopy(localContext,genericLocalContextClass) : new HashMap<>();
+        if (remoteContext != null) {
+            Class genericRemoteContextClass = remoteContext.getClass();
+            // TODO: 01.06.2021 redo
+            //this.remoteContext = helper.deepCopy(remoteContext, genericRemoteContextClass);
+        } else {
+            //this.remoteContext = new HashMap<>();
+        }
+        if (localContext != null) {
+            Class genericLocalContextClass = localContext.getClass();
+            // TODO: 01.06.2021 redo
+            //this.localContext = helper.deepCopy(localContext, genericLocalContextClass);
+        } else {
+            //this.localContext = new HashMap<>();
+        }
+
         initialized = true;
 
-        // TODO: 25.05.2021 emit
-        //emit(this, CONTEXT_INITIALIZED, this.resolve());
+        Object resolve = resolve();
+        List<Object> objects = new ArrayList<>();
+        objects.add(CONTEXT_INITIALIZED);
+        objects.add(resolve);
+
+        waitForIt.emit(this, CONTEXT_INITIALIZED, objects);
+    }
+
+
+    @Override
+    public boolean set(String key, Object value, boolean local) {
+        //checking incoming type
+        JsonElement jsonValue = null;
+        if(value instanceof JsonElement){
+            jsonValue = (JsonElement) value;
+        }else if(value instanceof String){
+            JsonParser parser = new JsonParser();
+            String modifyValue = ((String)value).replaceAll(" ",".");
+            jsonValue = parser.parse(modifyValue);
+        }
+
+        ensureInitialized();
+
+        JsonElement context = local ? localContext : remoteContext;
+        // TODO: 01.06.2021 add "before"
 
+        helper.setKeyToValue(key, jsonValue, context);
+
+        Object updated = this.resolve();
+
+        // TODO: 01.06.2021 add "if"
+
+        // TODO: 01.06.2021 understand the data type  emit(,,?)
+        waitForIt.emit(this, CONTEXT_CHANGED, null);
+        return true;
+
+    }
+
+    @Override
+    public Object resolve() {
+        ensureInitialized();
+        //return objects.deepClone(mutableResolve());
```

Comment:
What's this bit doing?

Reply:
it is the same logic as "**objects.deepClone(mutableResolve()**);"
![photo_2021-06-03_11-32-37](https://user-images.githubusercontent.com/28364912/120614337-c9b5bc80-c45f-11eb-9280-b983d9798a5c.jpg)
![photo_2021-06-03_11-33-47](https://user-images.githubusercontent.com/28364912/120614348-ccb0ad00-c45f-11eb-87fa-02e0dec6d4e9.jpg)